src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -703,19 +703,42 @@ TEST_F(TrajectoryPlannerTest, dynamics_dont_exceed_maximums_curved_path)
 
     for (float vel : velocity)
     {
-        EXPECT_TRUE(vel <= path_parameters.max_allowable_speed * 1.2 &&
-                    vel >= path_parameters.initial_speed);
+        EXPECT_TRUE(vel <= path_parameters.max_allowable_speed);
     }
 
     EXPECT_NEAR(velocity.back(), path_parameters.final_speed, 0.1);
 
     EXPECT_FLOAT_EQ(velocity.front(), path_parameters.initial_speed);
 
-    for (double acc : acceleration)
+    unsigned int i = 0;
+    // Acceleration phase coming up to a curve that must be slowed down for
+    while (fabs(acceleration[i] - path_parameters.max_allowable_acceleration) < 0.1)
+    {
+        EXPECT_NEAR(acceleration[i], path_parameters.max_allowable_acceleration, 0.001);
+        i++;
+    }
+    i++;  // Skip the transition acceleration from positive to negative
+
+    // Slowing down for a curve
+    while (fabs(acceleration[i] - path_parameters.max_allowable_acceleration) < 0.1)
+    {
+        EXPECT_NEAR(acceleration[i], -path_parameters.max_allowable_acceleration, 0.001);
+        i++;
+    }
+
+    i = 225;  // Skip the acceleration ramp-up phase
```

Comment:
`225` seems a bit magic. Can you do something similar to the other loops like

`while (fabs(acceleration[i] - path_parameters.max_allowable_acceleration) >= 0.1) {i++;}`
to skip the ramp up phase? that way if things change we don't need to change the magic number `225`

Reply:
Replaced i = 255 with a loop that checks until the trajectory is out of the section where acceleration is defined by path curvature